src/commands/serve.rs

```diff
@@ -144,7 +144,7 @@ async fn serve_static(req: actix_web::HttpRequest) -> actix_web::HttpResponse {
 }
 
 #[actix_web::main]
-pub async fn serve(bind_address: &str, port: &str) -> std::io::Result<()> {
+pub async fn serve(bind_address: &str, port: u16) -> std::io::Result<()> {
```

Comment:
We should change this to Option<u16>, if the port is specified at command line then we must use the port or error out. Only if post is not specified on cli we have to do port checking thing. Some(8000) should mean port 8000 was specified on cli. None to mean no port specified and find the first port that is available.